src/librustc/hir/mod.rs

```diff
@@ -925,7 +928,27 @@ pub struct BodyId {
     pub node_id: NodeId,
 }
 
-/// The body of a function or constant value.
+/// The body of a function, closure, or constant value. In the case of
+/// a function, the body contains not only the function body itself
+/// (which is an expression), but also the argument patterns, since
+/// those are something that the caller doesn't really care about.
+///
+/// Example:
+///
+/// ```rust
```

Comment:
conventionally the explicit `rust` here is only used in markdown files, not source files